code/common/compress.q

```diff
@@ -173,8 +173,8 @@ compress:{[filetoCompress;algo;blocksize;level;sizeuncomp]
                 $[not 0=algo;statstab ,: (filetoCompress;algo;(-21!filetoCompress)`compressedLength;sizeuncomp);statstab ,: (filetoCompress;algo;comprL;sizeuncomp)]];
             [$[(not count (-21!compressedFile));
 		[.lg.o[`compression; "Failed to compress file ",(string filetoCompress)];hdel compressedFile];
-		.lg.o[`compression; $[algo=0;"Decompressed ";"Compressed "], "file ", (string compressedFile), " doesn't match original. Deleting new file"]; hdel compressedFile]]]
+		.lg.o[`compression; $[algo=0;"Decompressed ";"Compressed "], "file ", (string compressedFile), " doesn't match original. Deleting new file"]; hdel compressedFile;]]]
```

Comment:
Use square brackets for `string`. If you were feeling particularly cheeky you could do `$[algo=0;"Dec";"C"]ompressed`

Reply:
Even better, you could do something that'll work e.g. `$[algo=0;"Dec";"C"],"ompressed"`

Reply:
@ThomasSmyth  I'm still a q newbie. Can I ask why you would want to write it that way? Is there some efficiency to be gained besides optimizing to reduce character count?

Reply:
Q programmers often strive for terseness in their code and this case is no different. My suggestion does not improve the efficiency, but will avoid the unnecessary duplication of strings.

I would argue that by making code more terse it can be made neater and cleaner. This should benefit those who have to read it later and can help simplify later maintenance of the file.

Reply:
Thank you very much! Reading these commit code reviews are invaluable to me.